pulsar-client/src/main/java/org/apache/pulsar/client/api/ReaderConfig.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api;
+
+import java.io.Serializable;
+
+public interface ReaderConfig<M extends Message> extends Serializable {
```

Comment:
It was a bit of a process as I was learning the code. I will convert this back.

Reply:
OK, I took a closer look. The implementation on the typed versions is based on wrapping the original client and providing a codec/schema concept at the outermost layer. Because the message listeners are inside the configuration objects, the configuration class has to be typed, I then had to be able to make a MessageListener<T> look like a MessageListener<byte[]> so it could be passed to the wrapped consumer, a Consumer<byte[]> (ConsumerImpl), which would require a ConsumerConfiguration<byte[]> and a Listener<byte[]>. I extracted the interface with the associated getters, since that is all that is used throughout the implementation. With this interface I could create an adapter from ConsumerConfiguration<T> (with a codec/schema) to a ConsumerConfiguration<byte[]> to successfully wrap the base implementation. Reader is the same issue. I would love to be able to deprecate the setListener method in the Configuration objects and allow someone to subscribe directly off the PulsarClient with MessageListener, I feel like that would be a more natural usage.